internal/controller/sink.go

```diff
@@ -263,3 +325,159 @@ func (c *configSink) newHTTPRoute(dst *gwapiv1.HTTPRoute, aiGatewayRoute *aigv1a
+									Args: []string{
+										"-configPath", "/etc/ai-gateway/extproc/" + expProcConfigFileName,
+										"-logLevel", "info",
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{Name: "config", MountPath: "/etc/ai-gateway/extproc"},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{
+								{
+									Name: "config",
+									VolumeSource: corev1.VolumeSource{
+										ConfigMap: &corev1.ConfigMapVolumeSource{
+											LocalObjectReference: corev1.LocalObjectReference{Name: extProcName(aiGatewayRoute)},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			updatedSpec, err := c.mountBackendSecurityPolicySecrets(&deployment.Spec.Template.Spec, aiGatewayRoute)
+			if err == nil {
+				deployment.Spec.Template.Spec = *updatedSpec
+			}
+			applyExtProcDeploymentConfigUpdate(&deployment.Spec, aiGatewayRoute.Spec.FilterConfig)
+			_, err = c.kube.AppsV1().Deployments(aiGatewayRoute.Namespace).Create(ctx, deployment, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create deployment: %w", err)
+			}
+			c.logger.Info("Created deployment", "name", name)
+		} else {
+			return fmt.Errorf("failed to get deployment: %w", err)
+		}
+	} else {
+		updatedSpec, err := c.mountBackendSecurityPolicySecrets(&deployment.Spec.Template.Spec, aiGatewayRoute)
+		if err == nil {
+			deployment.Spec.Template.Spec = *updatedSpec
+		}
+		applyExtProcDeploymentConfigUpdate(&deployment.Spec, aiGatewayRoute.Spec.FilterConfig)
+		if _, err = c.kube.AppsV1().Deployments(aiGatewayRoute.Namespace).Update(ctx, deployment, metav1.UpdateOptions{}); err != nil {
+			return fmt.Errorf("failed to update deployment: %w", err)
+		}
+	}
+
+	// This is static, so we don't need to update it.
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name,
+			Namespace:       aiGatewayRoute.Namespace,
+			OwnerReferences: ownerReferenceForAIGatewayRoute(aiGatewayRoute),
+			Labels:          labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{
+				{
+					Name:        "grpc",
+					Protocol:    corev1.ProtocolTCP,
+					Port:        1063,
+					AppProtocol: ptr.To("grpc"),
+				},
+			},
+		},
+	}
+	if _, err = c.kube.CoreV1().Services(aiGatewayRoute.Namespace).Create(ctx, service, metav1.CreateOptions{}); client.IgnoreAlreadyExists(err) != nil {
+		return fmt.Errorf("failed to create Service %s.%s: %w", name, aiGatewayRoute.Namespace, err)
+	}
+	return nil
+}
+
+// mountBackendSecurityPolicySecrets will mount secrets based on backendSecurityPolicies attached to AIServiceBackend.
+func (c *ConfigSink) mountBackendSecurityPolicySecrets(spec *corev1.PodSpec, aiGatewayRoute *aigv1a1.AIGatewayRoute) (*corev1.PodSpec, error) {
+	// Mount from scratch to avoid secrets that should be unmounted.
+	// Only keep the original mount which should be the config volume.
+	spec.Volumes = []corev1.Volume{spec.Volumes[0]}
```

Comment:
nit; then the original slice is not in use, so we don't need to allocate another slice again. instead we could reuse it
```suggestion
	// Mount from scratch to avoid secrets that should be unmounted.
	// Only keep the original mount which should be the config volume.
	spec.Volumes = spec.Volumes[:1]
```